### PR TITLE
feat(recoil): add application level state management for userPage

### PR DESCRIPTION
**Pull request by davidholyko**
#### Summary

Resolves: https://github.com/f1v/full-git-commit-history-app/projects/1#card-52579895 

- saves user repo data so that on lookup, doesn't re do all requests a second time

Known Issues:

- broke the Chart that is soon to be removed.

#### Checklist

- [x] Branch is in format `TYPE/scope/description`
- [x] PR title is in semantic format `TYPE(scope): description`
- [x] Commits Messages are in semantic format `TYPE(scope): description`
- [x] Has Summary
- [x] Has Labels

### CODE REVIEW

File: src/components/pages/UserPage.js [hunk 1]
```diff
@@ -3,11 +3,18 @@ import { Link, Redirect } from 'react-router-dom';
 import D3Chart from '../chart/D3Chart';
 import API from '../../utils/api';
 import PropTypes from 'prop-types';
+import { atom, useRecoilState } from 'recoil';
+
+const userState = atom({
```
Comment (trybick) [inline src/components/pages/UserPage.js]: nit - could move this to a new folder `recoil/atoms` and import userState

Also would `userRepoState` or something like that be more specific?

File: src/components/pages/UserPage.js [hunk 2]
```diff
@@ -34,16 +41,23 @@ export const UserPage = ({ match }) => {
...
+    const newUserRepos = Object.assign({}, userRepos);
+    Object.assign(newUserRepos, { [user]: repoData });
+    setUserRepos(newUserRepos);
   };
 
   useEffect(() => {
-    getData();
-    return () => {};
+    // only getData if its not already populated
+    if (!repos.length) {
```
Comment (trybick) [inline src/components/pages/UserPage.js:54]: I had to remove this if condition when testing fetching a second user. or does that work now?

File: src/components/pages/UserPage.js (hunk 1 above)
Comment (davidholyko) [reply src/components/pages/UserPage.js]: I do like userRepoState

File: src/components/pages/UserPage.js (hunk 2 above)
Comment (davidholyko) [reply src/components/pages/UserPage.js:54]: it works now

**Review by trybick — APPROVED** [review]

**Pull request merged**